src/npctalk.cpp

```diff
@@ -2255,6 +2256,14 @@ void dialogue::gen_responses( const talk_topic &the_topic )
             // TODO: Allow NPCs to break training properly
             // Don't allow them to walk away in the middle of training
             std::stringstream reasons;
+            int part;
+            vehicle *veh = g->m.veh_at( p->pos(), part );
```

Comment:
When you don't need the part index, don't request it (use the `map::veh_at` overload without the `part` parameter).